recipes/cassandra-cpp-driver/all/patches/2.16.1/fix-cmake.patch

```diff
@@ -0,0 +1,156 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c6150948..669eb086 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,6 +1,3 @@
+-cmake_minimum_required(VERSION 2.8.12)
+-project(cassandra C CXX)
+-
+ set(CASS_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+ set(CASS_SRC_DIR "${CASS_ROOT_DIR}/src")
+ set(CASS_INCLUDE_DIR "${CASS_ROOT_DIR}/include")
+@@ -81,7 +78,6 @@ endif()
+ # Dependencies
+ #------------------------
+ 
+-include(Dependencies)
+ include(ClangFormat)
+ 
+ #------------------------
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index d6627187..7e81b2a0 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -1,6 +1,50 @@
+ include(CheckSymbolExists)
+ include(CheckCXXSourceCompiles)
+ 
++#------------------------
++# Windows libraries
++#------------------------
++
++if(WIN32)
++  # Assign additional library requirements for Windows
++  set(CASS_LIBS iphlpapi psapi wsock32 crypt32 ws2_32 userenv version)
++endif()
++
++#------------------------
++# Libuv
++#------------------------
++
++if(NOT WIN32)
++  set(CMAKE_THREAD_PREFER_PTHREAD 1)
++  set(THREADS_PREFER_PTHREAD_FLAG 1)
++endif()
++
++find_package(Threads REQUIRED)
++set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
++set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
++if(NOT WIN32 AND ${CMAKE_VERSION} VERSION_LESS "3.1.0")
++  # FindThreads in CMake versions < v3.1.0 do not have the THREADS_PREFER_PTHREAD_FLAG to prefer -pthread
```

Comment:
🟨 CCI Team Question: Since we support 3.5+ do we need to keep moving this forward?

Reply:
up to contributors - it's not obligatory to keep backward compatibility for the ancient versions, but as long as it doesn't introduce too much burden and maintenance, it's okay to keep such things

Reply:
I think that's very well put. 💯